### PR TITLE
Move repository-s3 fixture tests to QA third party project

### PR DESCRIPTION
**Pull request by tlrx**
This commit moves the repository-s3 fixture test added in #29296 in a new `qa/third-party/amazon-s3` project. This new project allows the REST integration tests to be executed using the real S3 service when
all the required environment variables are provided. When no enviromnent variable is provided, the tests are executed using the fixture added in #29296.

The REST tests located in the plugin project now only verify that the plugin is correctly loaded.

The REST tests in the QA project have been adapted to allow a bucket name and a base path to be specified as env vars. This way it is possible to run the tests with different base paths (could be anything, like a CI job name or a branch name) without multiplicating buckets.

Related to #29349

### CODE REVIEW

File: qa/third-party/amazon-s3/build.gradle
Comment (tlrx) [inline qa/third-party/amazon-s3/build.gradle]: Note: the task is always created but not used when env vars are provided. I find it easier this way.

**Comment by tlrx** [comment]
@elastic/es-core-infra Can someone from the team can review this pull request please? Or maybe do you prefer to hold on this change until another major change is merged? (I'm fine with this but I'd like to know as I have few others PRs that are blocked by this one)

Note that this change should not impact the CI as it will use the fixture to run tests by default.

File: qa/third-party/amazon-s3/build.gradle
Comment (ywelsch) [inline qa/third-party/amazon-s3/build.gradle]: I wonder if we should log something if the fixture is not used, i.e., using external service (bucket + path).

**Review by ywelsch — CHANGES_REQUESTED** [review]
Recent discussions (Core/Infra meeting yesterday) have concluded that we should put QA projects that are really only testing a single plugin (and not the integration of many) into a subdirectory of that project and hence as a sub-project in Gradle. One of the biggest advantages is that this allows you to test all aspects related to that plugin with just one simple Gradle call. Examples for this are the CCR plugin (see https://github.com/elastic/elasticsearch/tree/ccr/x-pack/plugin/ccr) or the index lifecycle plugin.

The change looks good otherwise.

File: qa/third-party/amazon-s3/build.gradle
Comment (tlrx) [reply qa/third-party/amazon-s3/build.gradle]: I don't think we should log the bucket name, instead it will print a simple "Using an external service to test the repository-s3 plugin" message

**Comment by tlrx** [comment]
Thanks @ywelsch, I didn't know about this decision. The QA test is now located in the `plugins/repository-s3/qa/amazon-S3` project. I hooked up the check task so that executing `:plugin:repository-s3:check` also executes `:plugin:repository-s3:qa:amazon-s3:check`.

I had to adapt the project fileq of x-pack smoke tests so that they didn't try to include `:plugin:repository-s3:qa:amazon-s3` as if it was a real plugin.

File: x-pack/qa/smoke-test-plugins-ssl/build.gradle
Comment (ywelsch) [inline x-pack/qa/smoke-test-plugins-ssl/build.gradle]: I wonder if this is too much magic. An alternative might be to write `project.project(':plugins').getChildProjects.each { n, subproj -> ... }`
to just range over direct children of `plugins`.

**Review by ywelsch — APPROVED** [review]
LGTM

**Comment by rjernst** [comment]
> Recent discussions (Core/Infra meeting yesterday) have concluded that we should put QA projects that are really only testing a single plugin (and not the integration of many) into a subdirectory of that project and hence as a sub-project in Gradle.

While this was discussed, it was not concluded. Please don't do this until there is agreement on how (or if) it should be done.

**Comment by jasontedor** [comment]
> While this was discussed, it was not concluded. Please don't do this until there is agreement on how (or if) it should be done.

We had consensus on the point of tests for features/modules/plugins being next to the feature/module/plugin as sub-projects (e.g., we mentioned examples of CCR/index lifecycle management/reindex). There are some follow-up discussions needed related to the standing up of clusters with various features enabled/disabled, and related to security, but on the broader point of co-locating qa-style tests with the code, we did have consensus in the interest of optimizing workflow for developers on these.

It is apparent that you feel strongly otherwise so we can re-discuss this. However, I do not want to block @tlrx on this discussion where he was acting on guidance from his reviewer who saw the outcome of our previous discussion similarly. I will approve this PR as-is and if we come to a different decision when we discuss this again we can move the tests back (and we will not burden @tlrx with this work), this is okay.

This will be on the agenda for our next core/infra meeting.

**Review by jasontedor — APPROVED** [review]

File: x-pack/qa/smoke-test-plugins-ssl/build.gradle
Comment (tlrx) [reply x-pack/qa/smoke-test-plugins-ssl/build.gradle]: Thanks. This is indeed a good suggestion. I'll update this.

**Comment by tlrx** [comment]
Thanks a lot @ywelsch @jasontedor @rjernst! 

This pull request has been merged with the QA project being a sub-project of the `repository-s3` plugin. If this decision is revisited by the core/infra team, please don't hesitate to reach me out - I'd be happy to help to move them again.

**Pull request merged**
